### PR TITLE
only convert cljs map to object in native style prop vector

### LINKED ISSUE #67: accept non-object primitives in style vector
I have a component that accepts string shorthands in style vector, e.g. `{:style ["flex-1"]}`, but errors out with Helix since its wrapping all vector items in `primitive-obj`.

### PR DESCRIPTION
**Pull request by alidlorenzo**
fixes #67

### CODE REVIEW

File: src/helix/impl/props.cljc [hunk 1]
```diff
@@ -126,7 +126,7 @@
   [style]
   (cond
     ;; React Native allows arrays of styles
-    (vector? style) (into-js-array (map primitive-obj style))
+    (vector? style) (into-js-array (map #(if (map? %) (primitive-obj %) %) style))
     ;; when map, convert to an object w/ camel casing
     (map? style) (primitive-obj style)
```
Comment (Lokeh) [inline src/helix/impl/props.cljc:131]: can we swap the order of these while we're at it? I think it's more likely we'll hit the `map?` branch than the `vector?` branch typically.

**Comment by alidlorenzo** [comment]
I went ahead and made those fixes, but I'm not sure what exact contributing workflow is.

I don't use leiningen (been sticking with tools.deps), after installing it and trying to run tests - ran into issues because clojure and clojurescript were not in dependencies array. 

After getting tests to run, I couldn't get the actual tests to fail. Might have to do with the fact its a cljc file and cljs version of tests run with a different command?

Either way, if you could verify fixes and add contribution docs for future that would be great.

**Comment by Lokeh** [comment]
That's good feedback, I'll work on a document that lays out contribution guidelines and setup.

The workflow I typically use is to boot up shadow-cljs and connect my editor to it. There, I can run CLJ tests from the Clojure REPL.

To run the CLJS tests, I run the `:test` build and navigate to http://localhost:8021 that has the test runner UI, which hot reloads on changes.

This way, I can evaluate Clojure tests from the REPL and hot reload ClojureScript tests in the browser.

Hope that helps! Like I said, I'll do some work on cleaning up this workflow and documenting it since I would like it make it easier for people to contribute. Thanks again for the fix!

File: test/helix/impl/props_test.cljc [hunk 2]
```diff
@@ -105,7 +105,14 @@
 
      (t/is (eq (impl/native-props {:foo "bar"
                                    & nil})
-               #js {:foo "bar"}))))
+               #js {:foo "bar"}))
+
+     (t/is (eq (impl/native-props {:foo "bar"
+                                   & nil})
+               #js {:foo "bar"}))
```
Comment (Lokeh) [inline test/helix/impl/props_test.cljc:112]: Is this a duplicate of the test above?

Comment (alidlorenzo) [reply test/helix/impl/props_test.cljc]: my fault, missed that while I was playing around with test setup 

cleaned up that commit below

**Comment by Lokeh** [comment]
Looks great!

**Pull request merged**